independent-projects/arc/runtime/src/main/java/io/quarkus/arc/BindingsSource.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.arc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines a source of interception binding annotations. When this annotation
+ * is present on an {@link InterceptionProxy} parameter of a CDI producer method,
+ * the interception binding annotations present on the target class are ignored
+ * and instead are read from the class defined by this annotation.
+ * <p>
+ * Interceptor bindings for method named {@code M} with return type {@code R}
+ * and parameter types {@code P1, ..., Pn} on the target class are obtained from
+ * a method with an identical name, return type and parameter types declared
```

Comment:
How about generics? Imagine the target class `Foo` has a method `<T> void foo(List<T> list);`. I suppose that a method `void foo(List<T> list)` declared on `FooBindings1<T>` is matched but `<B> void foo(List<B> list)` declared on `FooBindings2`?

Reply:
Good point. I'm actually not sure how that would work today, I need to write a test.

Reply:
Changed the `BindingsDiscovery` class to not use `ClassInfo.method()`. Instead, we iterate through all methods of the class, comparing the method name, return type, parameter types, and the `static` flag. Also, discovering bindings on superclasses of the bindings source is now gone.

Reply:
> comparing ....return type, parameter types

By means of `Object#equals(Object)`?

Reply:
Yes, it's `Type.equals()`.